Assignment 2.Rmd

```diff
@@ -88,13 +90,66 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.widt
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
 
 
+```{r}
+
+#We can change the thickness of the edge according to the number of times a particular teacher has sent a particular student to the principal.
+
+edges.st <- EDGE %>% group_by(to) %>% summarize(sum(count))
+n.st <- unique(edges.st)
+edges.tch <- EDGE %>% group_by(from) %>% summarize(sum(count))
+n.te <- unique(edges.tch)
+names(edges.st) <- c("id", "count")
+names(edges.tch) <- c("id", "count")
+total <- rbind(edges.st, edges.tch)
+VERTEX2 <- full_join(VERTEX, total, by="id")
+g2 <- graph.data.frame(EDGE, directed = TRUE, vertices =  VERTEX2)
+
+
+#Now we can plot our graph using the force directed graphing technique - our old friend Fruchertman-Reingold!
+
+plot(g2,layout=layout.fruchterman.reingold)
+
+#There are many ways to change the attributes of the graph to represent different characteristics of the newtork. For example, we can color the nodes according to gender.
+
+plot(g2,layout=layout.fruchterman.reingold, vertex.color=VERTEX2$gender, vertex.size=VERTEX2$count+17, edge.arrow.size = 0.5)
+
+````
+
 
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
 
 Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
 
+```{r}
+#
+library(tidyr)
+library(dplyr)
+DT1 <- read.csv("hudk4050-classes.csv",header = TRUE)
+DT1$Name <- paste(DT1$First.Name, DT1$Last.Name)
+DT2_dirty <- DT1[,3:9]
+DT3 <- DT2_dirty %>% gather(classnum, classcode, `Class.1`, `Class.2`, `Class.3`, `Class.4`, `Class.5`, `Class.6`) %>% select(-c(classnum))
+DT3$classcode = gsub(" ", "", DT3$classcode)
+DT3 <- DT3 %>% filter(classcode != "HUDK4050") %>% filter(Name != "ZIMO CHEN")
+DT3$Count = 1
+DT3 <- DT3[which(DT3$classcode != ""),]
+DT4 <- DT3 %>% spread(classcode,Count)
+row.names(DT4) = DT4$Name
+DT4$name <- NULL
+DT4 = ifelse(is.na(DT4), 0, 1)
+DT5 = as.matrix(DT4)
+DT6 = DT5 %*% t(DT5)
+diag(DT6) <- NA
+g3 = graph.adjacency(DT6, mode = "undirected")
+plot(g3, layout=layout.fruchterman.reingold)
```

Comment:
It looks like you have not removed the class that all the students have in common (HUDK4050), so your network is a bit overconnected to be able to see any clusters. You also might want to try to reduce the font size so that you can more clearly see your network.